torchtune/modules/common_utils.py

```diff
@@ -163,3 +164,123 @@ def _register_reparametrize_state_dict_hooks(
     module._register_state_dict_hook(
         partial(hook, dtype=dtype, offload_to_cpu=offload_to_cpu)
     )
+
+
+@contextlib.contextmanager
+def use_persistent_kv_cache(model: nn.Module) -> Generator[None, None, None]:
```

Comment:
Since the more common use-case is that the KV cache WILL be used after it's setup, I might consider a paradigm shift. 

Instead of `use_kv_cache`, something like `disable_kv_cache` that temporarily disables the kv_cache for use in PPO. 

This also means that our current APIs still hold true.